component/azstorage/azstorage.go

```diff
@@ -105,6 +106,38 @@ func (az *AzStorage) Configure(isParent bool) error {
 	return nil
 }
 
+// Auto-Detect Account Type is HNS or FNS
+func (az *AzStorage) DetectAccountType(opt AzStorageOptions) {
+	auth := getAzAuth(az.stConfig.authConfig)
+	if auth == nil {
+		log.Err("DetectAccountType : Failed to get auth type %s", az.stConfig.authConfig.AuthMode)
+		return
+	}
+
+	serviceClient, err := auth.getServiceClient(&az.stConfig)
+	if err != nil {
+		log.Err("DetectAccountType : Failed to get service client %s", err.Error())
+		return
+	}
+	containerClient := serviceClient.(*service.Client).NewContainerClient(az.stConfig.container)
+
+	accountInfo, err := containerClient.GetAccountInfo(context.Background(), nil)
+	if err != nil {
+		log.Err("DetectAccountType : Failed to get account info %s", err.Error())
+		return
+	}
+	if *accountInfo.IsHierarchicalNamespaceEnabled {
+		az.stConfig.authConfig.AccountType = EAccountType.ADLS()
+		opt.Endpoint = fmt.Sprintf("%s.dfs.core.windows.net", opt.AccountName)
```

Comment:
If some private endpoint is set in config, then this logic would override it to *.core.windows.net which is incorrect